oxigen_stock_barcodes_gs1/wizard/stock_barcodes_read.py

```diff
@@ -37,12 +55,25 @@ def process_barcode(self, barcode):
             # try the AI 240 'Additional product identification assigned
             # by the manufacturer'.
             product_barcode = barcode_decoded.get("240", False)
-        lot_barcode = barcode_decoded.get("10", False)
         product_qty = barcode_decoded.get("37", False)
         if product_barcode:
-            product = self.env["product.product"].search(
-                self._barcode_domain(product_barcode)
+            self.env.cr.execute(
```

Comment:
Before the query, you should ensure that all fields involved are written to the database, add:
```python
# Flushing the relevant fields of the involved models
self.env['product.product'].flush(['barcode'])
self.env['product.template'].flush(['company_id'])
```